### DISCUSSION TITLE
Strategy for performance far and near zoom ST_AsMVT / Pmtiles / Function x,y,z

### DISCUSSION THREAD

**Discussion by IncidGeo** [maplibre/martin#1192] [General]
Hi,
Martin is great and fast to share tiles for near zoom tiles application in conjunction with Postgis ST_AsMVT function but not so good (the way i use it) for far zoom where a simplification of geometries would be welcome.
In opposite producing MBtiles with tippecanoe can greatly reduce the size of the geometries for far zoom but create an enormous quantity of tiles for near zoom and of course lost the simplicity of Postgres.
I was thinking of generating 2/3 tables in Postgres corresponding to 2/3 step of zooming and simplification levels of geometries with ST_Simplify function, then use the function x,y,z to select the corresponding table depending of z variable.
Is it a good idea ?  What is your strategy folks to preserve level of precision at near zoom but not on far zoom ?
Are MBtiles faster than ST_AsMVT in your application ?

**Comment by nyurik** [comment]
Martin is actually a very simple gateway to whatever method of producing tiles you use.  If your data is in Postgres, you essentially rely on PG to do all the geometry processing. PMTiles is always a much faster choice simply because all the processing has already been done before, possibly from PostgreSQL queries, or using Planetiler, or even using `martin-cp` to copy it from any of Martin-supported sources.  PostgreSQL gives you a good path for dynamically supported data. PMtiles/mbtiles are much faster but static.

In general, if your data is frequently changing, you may want to use something like materialized views for low zooms that you refresh once in a while.  And then you can have a PG function that goes to different tables/views/mat-views depending on the zoom requested.

**Reply by IncidGeo** [reply]
"In general, if your data is frequently changing, you may want to use something like materialized views for low zooms that you refresh once in a while. And then you can have a PG function that goes to different tables/views/mat-views depending on the zoom requested." 
Thank you nyurik i'll try this way. I've also change in my first post "but not so good" to "but not so good (the way i use it)"
